### PR TITLE
Fixed & Added bet Streaks/Stats

### PR DESCRIPTION
**Pull request by Alec-Fox**
fixed Stats/Streaks functions & added them to message in outputUserGamblingData

### CODE REVIEW

File: gambling.js [hunk 1]
```diff
@@ -45,6 +80,44 @@ exports.dischargeScrubBubble = function (userID, botSpam) {
...
+    if(ledger[userID] && ledger[dispatchee] && ledger[userID].armySize >= amount){
+        addToArmy(userID, amount, 'take');
+        addToArmy(dispatchee, amount, 'add');
+
+        description = '<@!' + dispatchee + '>'+ ' '+ amount +  ' Scrubbing Bubble' + maybeGetPlural(amount) + ' have been dispatched to you by' + '<@!' + userID + '>';
+        util.sendEmbedMessage(null, description);
+    }
+    
+}
+/**
```
Comment (kanagan123) [inline gambling.js]: add description of function above @params as seen above in various other function's JSDoc

File: gambling.js [hunk 2]
```diff
@@ -45,6 +80,44 @@ exports.dischargeScrubBubble = function (userID, botSpam) {
     const title = dropped + ' Scrubbing ' + msg + ' arrived for duty!';
     util.sendEmbedMessage(title, null, c.BUBBLE_IMAGES[droppedImg-1]);
 }
+/**
```
Comment (kenthannigan) [inline gambling.js]: add description of function above @params as seen above in various other function's JSDoc

File: gambling.js [hunk 3]
```diff
@@ -45,6 +80,44 @@ exports.dischargeScrubBubble = function (userID, botSpam) {
...
+        util.sendEmbedMessage(null, description);
+    }
+    
+}
+/**
+ * @param {String} mode - Decides which function to call
+ * @param {Number} userID - the id of the user calling command
+ * @param {String} args - text after command
+ */
+exports.checkNumber = function(mode, userID, args){
```
Comment (kanagan123) [inline gambling.js]: give this function a more descriptive name for what it is doing. checkNumber is very generic and does not make sense in the context of calling !dispatch.

File: gambling.js [hunk 4]
```diff
@@ -179,11 +230,11 @@ function addToGamblingStreaks(currentStreak, highestStreak, oppositeStreak, user
  * @param {String} user - the user to add stats to
  */
 function addToGamblingStats(outcome, amount, user) {
-    var plural = 'Losses';
+    var plural = 'Lost';
     var opposite = 'Won';
 
     if (outcome === 'Won') {
-        plural = 'Wins';
+        plural = 'Won';
```
Comment (kanagan123) [inline gambling.js:207]: Why set plural to 'Won' when outcome already holds the value 'Won'? You now have two variables holding the same value.

File: gambling.js [hunk 5]
```diff
@@ -179,11 +230,11 @@ function addToGamblingStreaks(currentStreak, highestStreak, oppositeStreak, user
  * @param {String} user - the user to add stats to
  */
 function addToGamblingStats(outcome, amount, user) {
-    var plural = 'Losses';
+    var plural = 'Lost';
```
Comment (kanagan123) [inline gambling.js:203]: I don't see the need for plural now. None of your variables names need it. 
```
                          streakWon         highestStreakWon     highestStreakLost
 addToGamblingStreaks('streak'+plural, 'highestStreak'+plural, 'streak'+opposite, user);
		scrubsWon
 ledger[user]['scrubs'+outcome] += amount;
		totalWon
 ledger[user]['total'+outcome]++;
```

File: gambling.js [hunk 6]
```diff
@@ -17,22 +18,56 @@ exports.exportLedger = function() {
...
+}
+
+/**
+ * Adds the given number of Scrubbing Bubbles to the provided user's army.
+ * 
+ * @param {String} userID - id of the user to add to
+ * @param {Number} amount - amount to add
+ * @param {String} addTake - Whether to Add or Take from Army
+ */
+function addToArmy(userID, amount, addTake) {
```
Comment (kanagan123) [inline gambling.js]: First of all why is a function called addToArmy able to take from an army? second of all why are you passing a string in, when you don't actually need to use that string in the function. you might as well be using a boolean. this is known as using flags and generally considered very bad design. conditional complexity could probably be improved.

**Comment by kanagan123** [comment]
please fix this

**Pull request closed without merging**